src/JsonSchema/Constraints/Factory.php

```diff
@@ -73,9 +73,43 @@ public function __construct(
         UriRetrieverInterface $uriRetriever = null,
         $checkMode = Constraint::CHECK_MODE_NORMAL
     ) {
+        // set provided config options
+        $this->setConfig($checkMode);
+
         $this->uriRetriever = $uriRetriever ?: new UriRetriever;
         $this->schemaStorage = $schemaStorage ?: new SchemaStorage($this->uriRetriever);
-        $this->checkMode = $checkMode;
+    }
+
+    /**
+     * Set config values
+     *
+     * @param int $checkMode Set checkMode options
+     * @param bool $enable Whether to enable or disable the provided options. If null, overwrites checkMode.
+     */
+    public function setConfig($checkMode = Constraint::CHECK_MODE_NORMAL, $enable = null)
```

Comment:
I find this a little confusing. If I pass in $checkMode but $not enable, then I overwrite the whole thing. Otherwise, you either add or remove the flag, dependent on $enable. Do you really think people are going to be busily setting and unsetting specific flags over the course of a program? Wouldn't it be simpler to just drop the $enable arg and just make this a simple set like you have on line 92? If someone want to unset a flag, they would do so by just setting the exact config they want and blowing away the old config...